File: org.summer.view.widget/src/org/summer/view/widget/TreeWalkPriority.java
package org.summer.view.widget;
/// <summary>
    ///     Enum specifying whether visual tree needs
    ///     to be travesed first or the logical tree
    /// </summary> 
    /*internal*/ public enum TreeWalkPriority
    { 
        /// <summary> 
        ///     Traverse Logical Tree first
        /// </summary> 
        LogicalTree,

        /// <summary>
        ///     Traverse Visual Tree first 
        /// </summary>
        VisualTree 
    } 

File: summer/WebContent/js/system/windows/TreeWalkPriority.js
/**
 * From DescendentsWalkerBase
 * TreeWalkPriority
 */
define(["dojo/_base/declare", "system/Type"], function(declare, Type){
	
    var TreeWalkPriority = declare("TreeWalkPriority", Object, { 
    });	
    
    /**
     * Traverse Logical Tree first
     */
    TreeWalkPriority.LogicalTree = 0;

    /**
     *  Traverse Visual Tree first 
     */
    TreeWalkPriority.VisualTree=1; 

//    Visibility.Type = new Type("Visibility", Visibility, [Object.Type]);
    return TreeWalkPriority;
});
